obspy/core/event/base.py

```diff
@@ -557,6 +561,9 @@ def __init__(self, id=None, prefix="smi:local",
             self.fixed = False
             self._prefix = prefix
             self._uuid = str(uuid4())
+        elif isinstance(id, ResourceIdentifier):
+            self.__dict__ = id.__dict__
```

Comment:
maybe copy this or rather `self.__dict__.update(id.__dict__)` - this might result in strange and hard to track errors otherwise.